### PR TITLE
Add new spec sys_block_queue_max_segment_size

### PR DESCRIPTION
**Pull request by TZ3070**
### All Pull Requests:

Check all that apply:

* [x] Have you followed the guidelines in our Contributing document, including the instructions about commit messages?
* [x] No Sensitive Data in this change?
* [ ] Is this PR to correct an issue?
* [ ] Is this PR an enhancement?
* [x] Need backport to `3.0_egg`? Yes, refer to [RPM/Egg Delivery](https://github.com/RedHatInsights/insights-core/blob/master/CONTRIBUTING.md#rpmegg-delivery) to open a new PR.
* [ ] Is this a backport from `master`? Yes, this is a backport of #PR-ID

### Complete Description of Additions/Changes:

- Add new spec sys_block_queue_max_segment_size
- Add new parser MaxSegmentSize
- Jira: RHINENG-22150

## Summary by Sourcery

Add support for parsing the max_segment_size attribute of block devices by introducing a new MaxSegmentSize parser, registering its spec, updating default specs, adding corresponding tests, and documenting the parser.

New Features:
- Add MaxSegmentSize parser for sys_block_queue_max_segment_size
- Register new spec sys_block_queue_max_segment_size and include it in default specs

Enhancements:
- Generalize test constants to CONTENT_INVALID and CONTENT_EMPTY for reuse

Documentation:
- Document the MaxSegmentSize parser in the shared parsers catalog

Tests:
- Add tests for MaxSegmentSize covering valid, invalid, and empty file content

### CODE REVIEW

File: insights/specs/__init__.py [hunk 1]
```diff
@@ -803,6 +803,7 @@ class Specs(SpecSet):
     sssd_conf_d = RegistryPoint(multi_output=True)
     sssd_logs = RegistryPoint(multi_output=True, filterable=True)
     sys_block_queue_stable_writes = RegistryPoint(multi_output=True)
+    sys_block_queue_max_segment_size = RegistryPoint(multi_output=True)
```
Comment (xiangce) [inline insights/specs/__init__.py]: The order of these two specs are not in alphabet order, would you mind moving both of them correctly under the `swift_proxy_server_conf`? Thanks.

**Review by xiangce — COMMENTED** [review]
other than below tiny suggestion, it looks good to me

**Review by JoySnow — APPROVED** [review]
